plenum/server/consensus/checkpoint_service.py

```diff
@@ -307,22 +307,25 @@ def _audit_txn_by_pp_seq_no(audit_ledger: Ledger, pp_seq_no: int) -> (dict, int)
         return txn, seq_no
 
     def process_new_view_accepted(self, msg: NewViewAccepted):
-        if not self.is_master:
-            return
-        # do not update stable checkpoints if the node doesn't have this checkpoint
-        # the node is lagging behind in this case and will start catchup after receiving one more quorum of checkpoints
-        # from other nodes
-        cp = msg.checkpoint
-        if cp not in self._data.checkpoints:
-            return PROCESS, None
-
-        # 1. update shared data
-        self._mark_checkpoint_stable(cp.seqNoEnd)
-        self.set_watermarks(low_watermark=cp.seqNoEnd)
-
-        # 2. send NewViewCheckpointsApplied
+        if self.is_master:
+            cp = msg.checkpoint
+            # do not update stable checkpoints if the node doesn't have this checkpoint
+            # the node is lagging behind in this case and will start catchup after receiving
+            # one more quorum of checkpoints from other nodes
+            if cp not in self._data.checkpoints:
+                return
+            self._mark_checkpoint_stable(cp.seqNoEnd)
+            self.set_watermarks(low_watermark=cp.seqNoEnd)
+        else:
+            # TODO: This is kind of hackery, but proper way would require introducing more
+            #  messages specifically for backup replicas. Hope we can live with it for now.
+            self._data.waiting_for_new_view = False
+            self._data.pp_seq_no = 0
+            self.set_watermarks(low_watermark=0)
+            self._reset_checkpoints()
+            self._data.stable_checkpoint = 0
```

Comment:
Actually `self._checkpointer._remove_received_checkpoints(till_3pc_key=(self.viewNo, 0))` won't clear anything (probably it could clear received checkpoints from previous views, but we're planning to remove viewNo from checkpoints, don't we?). Probably it makes sense to call  `self._checkpointer._remove_received_checkpoints()` which will remove ALL received checkpoints by that backup. It might clear too much, but eventually we'll receive quorum of checkpoints anyways. Any feedback on that idea?